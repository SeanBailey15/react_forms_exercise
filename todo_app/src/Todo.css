.Todo {
  display: flex;
  margin-bottom: 10px;
  width: 75vw;
}

.Todo-div {
  margin-left: 10px;
  border: 1px solid black;
  border-radius: 10px;
  width: 65vw;
  padding-left: 10px;
  background-color: paleturquoise;
}

.Todo-btn {
  font-weight: bolder;
  border: 3px solid black;
  border-radius: 10px;
  width: 30px;
  height: 30px;
  align-self: center;
  background-color: darkred;
  color: white;
}

.Todo-content {
  font-size: larger;
  font-weight: bolder;
}
