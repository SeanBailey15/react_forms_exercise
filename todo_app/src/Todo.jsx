import React from "react";
import './Todo.css'

const Todo = ({id, content, removeTodo}) => {
    const handleRemove = () => {
        removeTodo(id)
    };

    return (
        <div className="Todo">
            <button className="Todo-btn" type="submit" onClick={handleRemove} style={{display: "inline-block"}}>X</button>
            <div className="Todo-div" data-testid="todo">
                <p className="Todo-content">{content}</p>
            </div>
        </div>
    )
}

export default Todo