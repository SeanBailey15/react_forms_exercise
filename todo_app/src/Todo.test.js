import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

it("renders without crashing", () => {
  render(<Todo content={"Wash the car"} />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Todo content={"Wash the car"} />);
  expect(asFragment()).toMatchSnapshot();
});
