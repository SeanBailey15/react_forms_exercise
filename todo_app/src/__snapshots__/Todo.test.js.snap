// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <button
      class="Todo-btn"
      style="display: inline-block;"
      type="submit"
    >
      X
    </button>
    <div
      class="Todo-div"
      data-testid="todo"
    >
      <p
        class="Todo-content"
      >
        Wash the car
      </p>
    </div>
  </div>
</DocumentFragment>
`;
